justfile

```diff
@@ -113,13 +118,18 @@ protoc:
     rm protoc.zip
   fi
 
+
 # Installs python dependencies for testing
-pytest-setup:
-	poetry -C tests install
+venv:
+  python3 -c "import virtualenv" || python3 -m pip --quiet install virtualenv
+  python3 -m virtualenv {{VENV}} --quiet
+  {{VENV_BIN}}/python -m pip install poetry
+  {{VENV_BIN}}/poetry -C tests install
+
 
 # Runs pytest in the tests directory.
-pytest *args:
-	poetry -C tests run pytest --rootdir={{invocation_directory()}}/tests {{ if args == "" {'tests'} else {args} }}
+pytest *args: 
```

Comment:
previously this required poetry to be directly installed. Now it properly uses a virtual env.